src/caf/toolkit/translation.py

```diff
@@ -853,6 +853,37 @@ def pandas_vector_zone_translation(
     )
 
 
+def pandas_multi_column_translation(
+    vector: pd.DataFrame,
+    translation: pd.DataFrame,
+    from_col: str,
+    to_col: str,
+    factors_col: str,
+):
+    """
```

Comment:
I feel like this function is too specific to CAF.core to belong here.
I'd say either it should be generalised in-line with other functions in this module, or moved into core. (I'd rather it be generalised tbh)

I assumed it took a DF with the zoning system as the index and translated all other columns. Some questions:
- How do we know the df index is in the same order for the translation and the given vector?
- How is this different to just calling the vector translation on each column of `vector`?
  - Is it slower than the above method as you're using pandas methods rather than pure numpy?